### PR TITLE
Add a --motd option to print the report status to /etc/motd

### PR DESCRIPTION
**Pull request by roidelapluie**

### CODE REVIEW

**Comment by roidelapluie** [comment]
Result is:

```
###########################
# fna060.ant.exa.lab #
###########################

Last puppet run happened at 2016-03-14 11:00:04 +0100 in environment example.
The result of this puppet run was unchanged.
Events: Failure: 0. Success: 0. Total: 0.

```

**Comment by ripienaar** [comment]
I merged a few other things so all these need rebasing please

This is a good idea I need to give it a closer review. This code is quite dense I might throw in some new lines to improve reading a bit

Will check it out post rebase, ping me here once you updated

**Comment by roidelapluie** [comment]
rebased

**Comment by roidelapluie** [comment]
mmh please do not merge now

**Comment by roidelapluie** [comment]
I added some newlines

File: report_print.rb [hunk 1]
```diff
@@ -62,6 +62,40 @@ def print_report_summary(report)
...
+  header = "# #{report.host} #"
+  headline = "#" * header.size
+  motd << headline << header << headline << ''
+
+  motd << "Last puppet run happened at %s in environment %s." % [report.time, report.environment]
+
+  motd << "The result of this puppet run was "
+  case report.status
+  when 'failed'
+    motd.last << color(:red, report.status)
```
Comment (ripienaar) [inline report_print.rb]: I find this pattern of `motd.last` quite confusing.

A better option might be:

```
status_colors = {"failed" => :red.......}

motd << "The result of this puppet run was: %s." % color(status_colors[report.status], report.status)
```

this replaces the whole case statement and all this `.last << ...` stuff with something more obvious? What do you think?

Comment (ripienaar) [inline report_print.rb]: Could also be that writing to a file is better than maintaining the array? using tmpfile maybe?

Comment (roidelapluie) [inline report_print.rb]: I do not know how good Ruby it as deleting tempfiles in case of errors (e.g syntax error?)

Comment (roidelapluie) [inline report_print.rb]: Plus tmpfiles would be in /tmp which is typically in another filesystem that /etc. Looks like too much of an overhead to me

Comment (ripienaar) [inline report_print.rb]: temp files + atomic writes are a solved problem and only works on the same file system, there are libs for this.

anyway, its fine, if you could remove these `motd.last <<` stuff as above we can do the temp file later

**Comment by roidelapluie** [comment]
@ripienaar what do you think about this update?

File: report_print.rb [hunk 2]
```diff
@@ -41,7 +41,21 @@ def resources_of_type(report, type)
...
 def color(code, msg, reset=false)
-  colors = {:red => "[31m", :green => "[32m", :yellow => "[33m", :cyan => "[36m", :bold => "[1m", :reset => "[0m", :underline => "[4m"}
+  colors = {
+    :red       => "[31m",
+    :green     => "[32m",
+    :yellow    => "[33m",
+    :cyan      => "[36m",
+    :bold      => "[1m",
+    :underline => "[4m",
+    :reset     => "[0m",
```
Comment (roidelapluie) [inline report_print.rb]: @ripienaar I put the reset at the end so you can cat report_print.rb and not have everything underlined :)

**Comment by ripienaar** [comment]
nice, thanks :)

**Pull request merged**